fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/RecordMetaData.java

```diff
@@ -118,6 +119,7 @@ protected RecordMetaData(@Nonnull Descriptors.FileDescriptor recordsDescriptor,
         this.usesSubspaceKeyCounter = usesSubspaceKeyCounter;
         this.recordCountKey = recordCountKey;
         this.usesLocalRecordsDescriptor = usesLocalRecordsDescriptor;
+        this.recordTypesForIndex = new HashMap<>();
```

Comment:
Does this need to be a `ConcurrentHashMap`? Alternatively, this could be populated at meta-data build time (for every index), I suppose, and then it's immutable, so concurrency doesn't matter.

Reply:
@alecgrieser Fixing...